src/Nervos/CellDep.h

```diff
@@ -0,0 +1,39 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "OutPoint.h"
+#include "../proto/Nervos.pb.h"
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+
+namespace TW::Nervos {
+
+enum DepType { Code = 0, DepGroup = 1 };
+
+/// Nervos cell dep.
+struct CellDep {
+    OutPoint outPoint;
+    DepType depType;
+
+    /// Initializes a cell dep with a previous output and depType
+    CellDep(OutPoint outPoint, DepType depType) : outPoint(std::move(outPoint)), depType(depType) {}
```

Comment:
Pass-by-reference here?

Reply:
Hi @catenocrypt we had previously used pass-by-reference. But Milerius told to not use it in this comment: https://github.com/trustwallet/wallet-core/pull/2293#discussion_r892218089